### PR TITLE
actions: run: allow to start scripts with arguments

### PR DESCRIPTION
**Pull request by d4s**
Scripts passed to "Run" action are strings containing full path to script
with arguments. Debos fails to run it tr to start the whole string as a script.

Add possibility to start scripts with additional arguments.
Correct mount points for scripts with arguments in chrooted environment.

Signed-off-by: Denis Pynkin <denis.pynkin@collabora.com>

### CODE REVIEW

File: actions/run_action.go [hunk 1]
```diff
@@ -86,14 +90,19 @@ func (run *RunAction) doRun(context debos.DebosContext) error {
...
 		if run.Chroot {
-			cmd.AddBindMount(path.Dir(run.Script), "/script")
-			cmdline = []string{fmt.Sprintf("/script/%s", path.Base(run.Script))}
+			scriptpath, err := debos.GetCommandFromString(run.Script)
+			if err != nil {
+				return err
+			}
+			scriptpath = path.Dir(scriptpath)
+			cmd.AddBindMount(scriptpath, "/script")
+			cmdline = []string{strings.Replace(run.Script, scriptpath, "/script", 1)}
```
Comment (sjoerdsimons) [inline actions/run_action.go]: That will break if the directory name of the script and some other parameters overlap; please make sure you only replace the script name not the full command line

File: actions/run_action.go [hunk 2]
```diff
@@ -86,14 +90,19 @@ func (run *RunAction) doRun(context debos.DebosContext) error {
...
 		} else {
 			cmdline = []string{run.Script}
 		}
 		label = path.Base(run.Script)
 	} else {
-		cmdline = []string{"sh", "-c", run.Command}
+		cmdline = []string{run.Command}
```
Comment (sjoerdsimons) [inline actions/run_action.go]: Why this change?

File: commands.go [hunk 3]
```diff
@@ -66,6 +67,26 @@ func (w *commandWrapper) flush() {
 	w.out(true)
 }
 
+/*
+GetCommandFromString extracts command path from single string
+containing full path to command and arguments.
+*/
+func GetCommandFromString(command string) (string, error) {
```
Comment (sjoerdsimons) [inline commands.go]: Unless go has helpers just get the first word from the string rather then trying to be clever. The support for commands with spaces in them is impossible to end well (unless you enforce escaping).

File: commands.go [hunk 4]
```diff
@@ -111,8 +130,12 @@ func (cmd Command) Run(label string, cmdline ...string) error {
 			options = append(options, "--bind", b)
 
 		}
-		options = append(options, cmdline...)
 	}
+	if len(cmdline) == 1 {
+		// Single command OR command with options passed as single string
+		options = append(options, "sh", "-c")
```
Comment (sjoerdsimons) [inline commands.go]: I don't think this class should try to guess if it should use sh -c, the caller should do that.

File: actions/run_action.go (hunk 1 above)
Comment (d4s) [reply actions/run_action.go]: There is a limit '1' for replacement in `strings.Replace(..,..,1)`

File: actions/run_action.go (hunk 2 above)
Comment (d4s) [reply actions/run_action.go]: we need the same for scripts. Added shell below.
